client/src/components/Home.js

```diff
@@ -65,60 +64,93 @@ const Home = ({ user, logout }) => {
   const postMessage = (body) => {
     try {
       const data = saveMessage(body);
-
-      if (!body.conversationId) {
-        addNewConvo(body.recipientId, data.message);
-      } else {
-        addMessageToConversation(data);
-      }
-
-      sendMessage(data, body);
+      data.then(res => {
+        if (!body.conversationId) {
+          addNewConvo(body.recipientId, res.message);
+        } else {
+          addMessageToConversation(res, body.recipientId);
+        }
+        sendMessage(res, body);
+      });
     } catch (error) {
       console.error(error);
     }
   };
 
+
   const addNewConvo = useCallback(
     (recipientId, message) => {
-      conversations.forEach((convo) => {
+      let newState = [...conversations];
+      let currentConv = null
+      conversations.forEach((convo, index) => {
```

Comment:
Here you're creating a copy of the `conversations` array, but then modifying the original `conversations` array.